server/service/microsoft_mdm.go

```diff
@@ -1188,31 +1194,104 @@ func (svc *Service) GetMDMWindowsTOSContent(ctx context.Context, redirectUri str
 }
 
 // isTrustedRequest checks if the incoming request was sent from MDM enrolled device
-func (svc *Service) isTrustedRequest(ctx context.Context, reqSyncML *fleet.SyncML) error {
+func (svc *Service) isTrustedRequest(ctx context.Context, reqSyncML *fleet.SyncML, reqCerts []*x509.Certificate) error {
 	if reqSyncML == nil {
 		return fleet.NewInvalidArgumentError("syncml req message", "message is not present")
 	}
 
-	// TODO - Security checks that need to be implemented
-	// - TLS based auth
-	// - Device auth based on Source/LocURI DeviceID information (this should be present on Enrollment DB)
+	// Checking if calling request is coming from an already MDM enrolled device
+	deviceID, err := reqSyncML.GetSource()
+	if err != nil || deviceID == "" {
+		return fmt.Errorf("invalid SyncML message %w", err)
+	}
+
+	enrolledDevice, err := svc.ds.MDMWindowsGetEnrolledDeviceWithDeviceID(ctx, deviceID)
+	if err != nil || enrolledDevice == nil {
+		return errors.New("device was not MDM enrolled")
+	}
 
-	return nil
+	// Check if TLS certs contains device ID on its common name
+	if len(reqCerts) > 0 {
+		for _, reqCert := range reqCerts {
+			if strings.Contains(reqCert.Subject.CommonName, deviceID) {
+				return nil
+			}
+		}
+	}
+
+	// TODO: Latest version of the MDM client stack don't populate TLS.PeerCertificates array
```

Comment:
I am building my own windows mdm solution in rust. I am following your mdm development from a month or so.

From many days I am  stuck on 'why is the client not sending certificate'. I didn't know that latest mdm clients don't send certs. Can you link where did you get this info from. Thanks